### PR TITLE
Darwin support

### PR DESCRIPTION
**Pull request by 2010YOUY01**
For issue https://github.com/agroal/pgagroal/issues/325
This is an awesome project! I (and possibly other contributors) would like to do development on MacOS X, so I'm trying to make it build on MacOS X.
This patch is currently separated into small commits for review, and will be squashed later if it's possible to merge.

### [Fix linker in cmake.](https://github.com/agroal/pgagroal/commit/f6257afee85d8fb8ca81f0319fd0288c6433a29e)
The semantics of `check_c_compiler_flag(-Wl,-z,relro HAS_RELRO)` is to check if you pass `-Wl -z relro` to the compiler and if it will throw an error, the compiler will simply pass it to the linker and didn't actually check if linker will throw an error for `-z relro` option.  e.g. I tested `check_c_compiler_flag(-Wl,random,var FOO)` and `FOO` is set to true, so the current implementation might have false positive and make the project's cmake fail on MacOS X. Detail also discussed at https://github.com/agroal/pgagroal/issues/325#issuecomment-1470825193
### [find OpenSSL and headers on Darwin](https://github.com/agroal/pgagroal/commit/6e8744d3ab9d9d1dcb3d337a00dfc3fd3b57468e)
Setup include path for OpenSSL otherwise `openssl/ssl.h` can't be found when compiling.
Defined `_DARWIN_C_SOURCE` to locate system headers
Fixed `pgagroal_set_proc_title` under MacOS X, I tested it using `ps`, and works fine.
### [fix rlim_t type print issue.](https://github.com/agroal/pgagroal/commit/d7f7e8ac877496bb695ab49237bdef12681b3889)
`rlim_t` type seems might be either 4 bytes or 8 bytes on different architectures, and its 8 byte size in MacOS X caused problem.
```
 error: format specifies type 'long' but the argument has type 'unsigned long long' [-Werror,-Wformat]
      errx(1, "max_connections is larger than the file descriptor limit (%ld available)", flimit.rlim_cur - 30);
                                                                         ~~~              ^~~~~~~~~~~~~~~~~~~~
                                                                         %llu
```
 Check size at compile time seems hard to do, and since pgagroal is currently deployed ok and might be only used for development under MacOS X, here just simply did type casting to fix the error.
### [Darwin CI setup and formatting.](https://github.com/agroal/pgagroal/commit/8a11fc5f6988e69892de47e3905f5f960d19c317)
CI, AUTHOR file, and several formatting issues caught by [uncrustify.sh](https://github.com/agroal/pgagroal/blob/master/uncrustify.sh)

## Testing
Deployed on local machine `Darwin Kernel Version 19.6.0`, run several simple queries through `pgagroal` proxy, and tested the concurrent connection using `pgbench`, no error reported.

### CODE REVIEW

**Comment by jesperpedersen** [comment]
Nice !

Lets start by doing a squash and force push - plus remove unrelated changes. Feel free to open multiple pull requests...

**Comment by 2010YOUY01** [comment]
Squash and force pushed. I forget to mention all modifications are required for the project to build on MacOS X.

File: src/libpgagroal/configuration.c [hunk 1]
```diff
@@ -2629,7 +2629,6 @@ transfer_configuration(struct configuration* config, struct configuration* reloa
    // changes the pgagroal-cli probably will not be able to connect in any case!
    restart_string("unix_socket_dir", config->unix_socket_dir, reload->unix_socket_dir, false);
 
-
```
Comment (jesperpedersen) [inline src/libpgagroal/configuration.c:2632]: Lets remove this change from this pull request

File: src/libpgagroal/message.c [hunk 2]
```diff
@@ -1275,7 +1275,7 @@ ssl_read_message(SSL* ssl, int timeout, struct message** msg)
          err = SSL_get_error(ssl, numbytes);
          switch (err)
          {
-            case SSL_ERROR_NONE: 
+            case SSL_ERROR_NONE:
```
Comment (jesperpedersen) [inline src/libpgagroal/message.c:1278]: Lets revert this as well

File: src/libpgagroal/message.c [hunk 3]
```diff
@@ -1371,7 +1371,7 @@ ssl_write_message(SSL* ssl, struct message* msg)
 
          switch (err)
          {
-            case SSL_ERROR_NONE: 
+            case SSL_ERROR_NONE:
```
Comment (jesperpedersen) [inline src/libpgagroal/message.c:1374]: Lets revert this as well

**Review by jesperpedersen — CHANGES_REQUESTED** [review]

**Comment by jesperpedersen** [comment]
Just some white space changes removed in order to minimize the patch...

**Comment by 2010YOUY01** [comment]
Sure, removed. And thanks for reviewing!

**Comment by jesperpedersen** [comment]
Merged.

Thank you for your contribution ! It was very helpful !

Feel free to port the patch to the "sister" projects - pgmoneta and pgexporter :)

**Comment by 2010YOUY01** [comment]
I haven't looked into pgmoneta and pgexporter yet. Let's first create issues linked to this PR, so that it will be easier for other contributors to use this patch as a reference if they want to port it themselves. 
However, I have already ported and used pgprtdbg. It's very handy for understanding the underlying pg protocol communication of pgagroal, will send a patch soon!

**Comment by jesperpedersen** [comment]
@2010YOUY01 No problem !

**Pull request merged**